app/src/main/java/com/nervousfish/nervousfish/activities/WaitActivity.java

```diff
@@ -33,6 +34,7 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger("WaitActivity");
     private IServiceLocator serviceLocator;
+    private IDatabase database;
```

Comment:
Okay. However, I still don't think that following the Law of Demeter is bad practice to follow, see http://www.virtuouscode.com/2011/07/05/demeter-its-not-just-a-good-idea-its-the-law/.